src/django/api/views/v1/production_locations.py

```diff
@@ -26,39 +26,43 @@ def __init__(self, **kwargs):
         self.opensearch_service = OpenSearchService()
         self.opensearch_query_builder = ProductionLocationsQueryBuilder()
         self.opensearch_query_director = OpenSearchQueryDirector(
-                self.opensearch_query_builder
-            )
+            self.opensearch_query_builder
+        )
 
     @handle_errors_decorator
     def list(self, request):
         _, error_response = serialize_params(
             ProductionLocationsSerializer,
-            request.GET
+            request.GET,
         )
+
         if error_response:
-            return Response(
-                error_response,
-                status=status.HTTP_400_BAD_REQUEST
-            )
+            return Response(error_response, status=status.HTTP_400_BAD_REQUEST)
 
         query_body = self.opensearch_query_director.build_query(
-            request.GET
+            request.GET,
         )
         response = self.opensearch_service.search_index(
             OpenSearchIndexNames.PRODUCTION_LOCATIONS_INDEX,
-            query_body
+            query_body,
         )
         return Response(response)
 
     @handle_errors_decorator
-    def retrieve(self, request,  pk=None):
-        query_params = QueryDict('', mutable=True)
-        query_params.update({'os_id': pk})
-        query_body = self.opensearch_query_director.build_query(
-            query_params
-        )
+    def retrieve(self, _, pk=None):
+        query_params = QueryDict("", mutable=True)
+        query_params.update({"os_id": pk})
+        query_body = self.opensearch_query_director.build_query(query_params)
         response = self.opensearch_service.search_index(
             OpenSearchIndexNames.PRODUCTION_LOCATIONS_INDEX,
-            query_body
+            query_body,
         )
-        return Response(response)
+        locations = response.get("data", [])
+
+        if len(locations) == 0:
+            return Response(
+                data={"detail": "Production location not found!"},
```

Comment:
Could you please use the same error message as specified in [the new API spec](https://opensupplyhub.github.io/open-supply-hub-api-docs/) for GET api/v1/production-locations/{os_id} endpoint?
```
{
  "message": "The location with the given id was not found."
}
```

Reply:
Sure, updated the message.